Hunted Mobile/Hunted Mobile/Hunted Mobile/Service/Json/JsonConversionService.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Hunted_Mobile.Model.Response.Json;
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace Hunted_Mobile.Service.Json {
+    public abstract class JsonConversionService<NativeType, DataType> where DataType : JsonResponseData {
+        protected readonly JsonSerializerSettings serializerSettings = new JsonSerializerSettings {
+            NullValueHandling = NullValueHandling.Ignore,
+            DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate
+        };
+
+        public JsonConversionService() {
+        }
+
+        public virtual string ToJson(NativeType @object) {
+            throw new NotImplementedException();
+        }
+
+        public NativeType ToObject(string json) {
+            return ToObject(ConvertFromJson(json));
+        }
+
+        public NativeType[] ToObjects(string json) {
+
+            var jsonArray = ToArray(json);
+            NativeType[] objectArray = new NativeType[jsonArray.Count];
+
+            for(int i = 0; i < objectArray.Length; i++) {
+                objectArray[i] = ToObject(ConvertFromJson(jsonArray[i]));
+            }
+
+            return objectArray;
+        }
+
+        public NativeType[] ToObjects(DataType[] data) {
+            NativeType[] objects = new NativeType[data.Length];
+            for(int i = 0; i < objects.Length; i++) {
+                objects[i] = ToObject(data[i]);
+            }
+            return objects;
+        }
+
+        public abstract NativeType ToObject(DataType data);
+
+        protected ReadOnlyCollection<string> ToArray(string json) {
+            List<string> items = new List<string>();
+            foreach(var item in ToJArray(json)) {
+                items.Add(item.ToString());
+            }
+            return items.AsReadOnly();
+        }
+
+        protected JArray ToJArray(string json) {
+            return JArray.Parse(json);
+        }
+
+        protected JObject ToJObject(string json) {
+            if(json == null) {
+                return new JObject();
+            }
+            else return JsonConvert.DeserializeObject<JObject>(json, serializerSettings);
```

Comment:
```suggestion
            return JsonConvert.DeserializeObject<JObject>(json, serializerSettings);
```
De `else` is hier niet nodig.

Reply:
Ik wil hier geen probleem van maken ofzo maar er is een rede waarom ik dat wel altijd toevoeg. Stel dat dit een grotere functie was met meerdere if/else scopes. Als je overal expliciet in een if of else een return doet en je vergeet ergens een return statement, dan geeft de IDE een error omdat je methode niet altijd een waarde returned. Als je de else weg laat en je vergeet een return statement, dan vindt je IDE dat geen probleem en heb je wss een bug.

Maar aan de andere kant is je functie misschien wel te groot als je zo'n probleem tegenkomt.